admin/model/billmate/payment/modificator.php

```diff
@@ -0,0 +1,396 @@
+<?php
+
+/**
+ * Class ModelBillmatePaymentModificator
+ */
+class ModelBillmatePaymentModificator extends Model
+{
+    const MODIFICATION_FILE = 'bm_modification.xml';
+
+    const MODIFICATION_CODE = 'billmate_checkout';
+
+    const STATUS_ENABLED = 1;
+
+    /**
+     * ModelBillmatePaymentBmsetup constructor.
+     *
+     * @param $registry
+     */
+    public function __construct($registry)
+    {
+        parent::__construct($registry);
+        $this->load->model('billmate/payment/modificator');
+    }
+
+    /**
+     *
+     */
+    public function addModifications()
+    {
+        $modData = $this->getModificationData();
+        if ($modData) {
+            $this->model_setting_modification->addModification($modData);
+            $this->refresh();
+        }
+    }
+
+    public function deleteModifications()
+    {
+        $this->db->query(
+            "DELETE FROM `" . DB_PREFIX . "modification` WHERE `code` = '" . self::MODIFICATION_CODE . "'"
+        );
+        $this->refresh();
+    }
+
+    /**
+     * @return array
+     */
+    protected function getModificationData()
+    {
+        $modData = [];
+        $file = DIR_SYSTEM .  self::MODIFICATION_FILE;
+        if (is_file($file)) {
+            $xml = file_get_contents($file);
+            if ($xml) {
+                $dom = new DOMDocument('1.0', 'UTF-8');
+                $dom->loadXml($xml);
+                $name = $dom->getElementsByTagName('name')->item(0)->nodeValue;
+                $author = $dom->getElementsByTagName('author')->item(0)->nodeValue;
+                $version = $dom->getElementsByTagName('version')->item(0)->nodeValue;
+                $link = $dom->getElementsByTagName('link')->item(0)->nodeValue;
+
+                $modData = [
+                    'extension_install_id' => 0,
+                    'name'                 => $name,
+                    'code'                 => self::MODIFICATION_CODE,
+                    'author'               => $author,
+                    'version'              => $version,
+                    'link'                 => $link,
+                    'xml'                  => $xml,
+                    'status'               => self::STATUS_ENABLED
+                ];
+            }
+        }
+        return $modData;
+    }
+
+    /**
+     * @Note This function copies functionality from the native OC ControllerMarketplaceModification
+     * We must be care with future updates (will be waiting when OC separate logic on separated functions)
+     */
+    public function refresh()
```

Comment:
Indentation is wrong in this method.

A we also need to split this method into smaller chunks and methods. It is way to huge for getting an overview.